scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -37,6 +43,16 @@ def process_response(self, request, response, spider):
             if content_encoding:
                 encoding = content_encoding.pop()
                 decoded_body = self._decode(response.body, encoding.lower())
+                if self._max_size and len(decoded_body) > self._max_size:
+                    error_msg = ("Cancelling processing of %(url)s: "
+                                 "Uncompressed response size %(size)s larger "
+                                 "than max allowed size (%(maxsize)s).")
+                    error_args = {'url': response.url,
+                                  'size': len(decoded_body),
+                                  'maxsize': self._max_size}
+                    logger.error(error_msg, error_args)
+                    raise IgnoreRequest(error_msg % error_args)
```

Comment:
Not sure IgnoreRequest is the right thing to raise here

Reply:
A good question. I think IgnoreRequest is a right thing to raise, but it is better to create a subclass, so that users can catch this exception specifically, and so that it shows up under a different name in logs and stats.